pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -535,15 +532,16 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             }
         } else {
             actualAmountOut = amountOut;
-            tokenOut = token;
 
             if (callParams.isStaticCall == false) {
                 _sendTokenOut(callParams.sender, IERC20(token), actualAmountOut, callParams.wethIsEth);
             }
         }
 
         if (actualAmountOut < minAmountOut) {
-            revert IVaultErrors.AmountOutBelowMin(IERC20(tokenOut), actualAmountOut, minAmountOut);
+            IERC20 tokenOut = needToUnwrap ? underlyingToken : IERC20(token);
+
+            revert IVaultErrors.AmountOutBelowMin(tokenOut, actualAmountOut, minAmountOut);
```

Comment:
Hm.

I think there's a small bug in `_processTokenInExactOut`, and this asymmetry made me realize about it.
Two things:
- This is fine, but alternatively you could keep `tokenOut` as a local before the first `if (needToUnwrap)`, and assign it accordingly. That way you would save the branch here, which checks whether you need to unwrap for a second time in the function.
- For the event in `_processTokenInExactOut` I think we're always emitting the same token, no matter whether we're actually dealing with the wrapped or the underlying. I think we need to emit `tokenIn`, not just `token` in there (commenting here because it's outside the diff).

Reply:
Same as above. Yeah, it's an extra assignment in the normal path, but now you need a larger scope for underlying token and you need to check `needToUnwrap` twice. IMO the other option is simpler.